### PR TITLE
Use Facilities API for VA medical center selection in Caregiver application

### PR DESCRIPTION
**Pull request by longmd**
## Description
Currently we are getting the list of available preferred facilities from a static JSON file. This PR updates the preferred facility list to be fetched by way of the Facilities API. This will allow for the most accurate list of facilities to be presented to the applicant for selection.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#26246

## Testing done
- [x] Cypress testing

## Acceptance criteria
- [ ] Facilities are dynamically fetched after state selection and not pulled from vets-json-schema enums
- [ ] UI behaves and looks as it did when using the static list
- [ ] Errors fetching should be logged to Sentry

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/caregivers/config/chapters/veteran/vetMedicalCenter_api.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  LastTreatmentFacilityAPIUI,
+  PreferredFacilityAPIUI,
+} from 'applications/caregivers/definitions/UIDefinitions/veteranUI';
+
+const {
+  lastTreatmentFacility,
+  plannedClinic,
+} = fullSchema.properties.veteran.properties;
+const caregiverStates = constants.states50AndDC
+  .concat([{ label: 'Puerto Rico', value: 'PR' }])
```
Comment (Mottie) [inline src/applications/caregivers/config/chapters/veteran/vetMedicalCenter_api.js:14]: Seems to me that if we include Puerto Rico, we should also include Guam, U.S. Virgin islands, etc...?

File: src/applications/caregivers/config/form.js [hunk 2]
```diff
@@ -119,11 +120,19 @@ const formConfig = {
...
-        veteranInfoThree: {
-          path: 'vet-3',
+        veteranInfoThreeJSON: {
+          path: 'vet-3-json',
           title: 'VA medical center',
-          uiSchema: vetMedicalCenterPage.uiSchema,
-          schema: vetMedicalCenterPage.schema,
+          depends: formData => !formData['view:useFacilitiesAPI'],
+          uiSchema: vetMedicalCenterJsonPage.uiSchema,
+          schema: vetMedicalCenterJsonPage.schema,
```
Comment (Mottie) [inline src/applications/caregivers/config/form.js:128]: Without seeing these pages, would a Veteran know what to do with JSON data?

Also, the path names could maybe be improved? I'm not an IA person, but in the past, I've named the path similar to the page title.

**Review by Mottie — APPROVED** [review]

**Review by lzapson — APPROVED** [review]

File: src/applications/caregivers/config/form.js (hunk 2 above)
Comment (longmd) [reply src/applications/caregivers/config/form.js:128]: This is only temporary, as we will be doing a phased in approach. Once the phase in is complete and we're satisfied, all the naming conventions go back to as they are now and the current codebase that fetches the static JSON file will go away.

File: src/applications/caregivers/config/chapters/veteran/vetMedicalCenter_api.js (hunk 1 above)
Comment (longmd) [reply src/applications/caregivers/config/chapters/veteran/vetMedicalCenter_api.js:14]: There are no caregiver facilities on those islands... only PR. Ideally, Id love for this list to be completely generated dynamically with the API, but it doesn't allow you to just search by service. You have to search by service and location.

**Review by alexandec — APPROVED** [review]

**Pull request merged**